flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionService.java

```diff
@@ -233,15 +241,20 @@ public void isLeader() {
 		}
 	}
 
+	private void clearConfirmedLeaderInformation() {
+		confirmedLeaderSessionID = null;
+		confirmedLeaderAddress = null;
+	}
+
 	@Override
 	public void notLeader() {
 		synchronized (lock) {
 			if (running) {
 				issuedLeaderSessionID = null;
-				confirmedLeaderSessionID = null;
+				clearConfirmedLeaderInformation();
 
 				if (LOG.isDebugEnabled()) {
-					LOG.debug("Revoke leadership of {}.", leaderContender.getAddress());
+					LOG.debug("Revoke leadership of {}.", leaderContender);
 				}
```

Comment:
I guess we want to output rather `confirmedLeaderAddress`, not `leaderContender`

```
				if (LOG.isDebugEnabled()) {
				    LOG.debug("Revoke leadership of {}.", confirmedLeaderAddress);
				}

				clearConfirmedLeaderInformation();
```

Reply:
fair enough it could be `leaderContender.getDescription` which also includes leader address.

Reply:
I'll change it to `"Revoke leadership of {} ({}@{})", leaderContender, confirmedLeaderSession, confirmedLeaderAddress`.

Once we have `getDescription` (with #9821), I'll change `leaderContender` to `leaderContender.getDescription`.